src/test/scala/za/co/absa/hyperdrive/trigger/scheduler/sensors/recurring/RecurringSensorTest.scala

```diff
@@ -0,0 +1,79 @@
+
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.trigger.scheduler.sensors.recurring
+
+import java.util.concurrent
+
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.{eq => eqTo, _}
+import org.mockito.Mockito.{reset, when}
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
+import za.co.absa.hyperdrive.trigger.TestUtils.await
+import za.co.absa.hyperdrive.trigger.models._
+import za.co.absa.hyperdrive.trigger.models.enums.SensorTypes
+import za.co.absa.hyperdrive.trigger.persistance.DagInstanceRepository
+import za.co.absa.hyperdrive.trigger.scheduler.eventProcessor.EventProcessor
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+
+class RecurringSensorTest extends FlatSpec with MockitoSugar with Matchers with BeforeAndAfter {
+  private implicit val executionContext: ExecutionContextExecutor =
+    ExecutionContext.fromExecutor(concurrent.Executors.newSingleThreadExecutor())
+  private val eventProcessor = mock[EventProcessor]
+  private val dagInstanceRepository = mock[DagInstanceRepository]
+
+  before {
+    reset(eventProcessor)
+    reset(dagInstanceRepository)
+  }
+
+  "RecurringSensor.poll" should "do nothing when workflow is running" in {
+    // given
+    val triggeredBy = "user"
+    val properties = Properties(-1L, Settings(Map.empty, Map.empty), Map.empty)
+    val sensorDefinition = Sensor(-1L, SensorTypes.Recurring, properties)
+
+    when(dagInstanceRepository.hasRunningDagInstance(eqTo(sensorDefinition.workflowId))(any[ExecutionContext])).thenReturn(Future{true})
+
+    val underTest = new RecurringSensor(eventProcessor.eventProcessor(triggeredBy), sensorDefinition, executionContext, dagInstanceRepository)
+
+    // when
+    val result: Unit = await(underTest.poll())
+
+    // then
+    result shouldBe (): Unit
```

Comment:
Type annotation not required here.
Instead of checking the result (is a bit meaningless, since `result` will always be `Unit` when the code reaches this point), I would verify that `eventProcessor.eventProcessor` hasn't been called.

Reply:
1. In case of unit, I prefer type annotation. (If you insist, I will remove it)
2. Hmm so I wrote it there, but I think it is not necessary because it is there implicitly. (Should throw exception)

Reply:
Ah yes ok you are right. The same is also true for `result shouldBe()` I leave it to you